presto-main/src/main/java/com/facebook/presto/resourcemanager/DistributedResourceGroupInfoResource.java

```diff
@@ -135,9 +165,43 @@ private ResourceGroupInfo aggregateResourceGroupInfo(List<ListenableFuture<Resou
         return builder.build();
     }
 
-    private ListenableFuture<ResourceGroupInfo> getResourceGroupInfoFromCoordinator(UriInfo uriInfo,
-            InternalNode coordinatorNode)
-            throws IOException
+    private List<ResourceGroupInfo> aggregateRootGroups(List<ListenableFuture<List<ResourceGroupInfo>>> rootGroupsFutureList)
+            throws InterruptedException, ExecutionException
+
+    {
+        Iterator<ListenableFuture<List<ResourceGroupInfo>>> iterator = rootGroupsFutureList.iterator();
+        Map<ResourceGroupId, AggregatedResourceGroupInfoBuilder> groupAggrators = new HashMap<>();
+        // iterator through all the root resource groups from all of the coordinators
+        // and aggregate each resource groups
+        while (iterator.hasNext()) {
+            try {
+                List<ResourceGroupInfo> rootGroups = iterator.next().get();
+                Iterator<ResourceGroupInfo> rootGroupsItor = rootGroups.iterator();
+                while (rootGroupsItor.hasNext()) {
```

Comment:
What about converting this to a `for`?
That way we can get rid of `ResourceGroupInfo rootGroup = rootGroupsItor.next();` and have it directly on the loop declaration

```
 for (ResourceGroupInfo rootGroup : rootGroups) {
   ResourceGroupId gid = rootGroup.getId();
   // ... Rest of the code
```

Reply:
actually `foreach` loop uses an iterator behind the scenes. As long as it uses an iterator, it would be good.

Reply:
I changed both iterators to the foreach loop